contracts/src/unruggable_memecoin.cairo

```diff
@@ -103,11 +115,19 @@ mod UnruggableMemecoin {
         initial_recipient: ContractAddress,
         name: felt252,
         symbol: felt252,
-        initial_supply: u256
+        initial_supply: u256,
+        factory_address: ContractAddress,
```

Comment:
putting those parameters as constructors args is obviously much more flexible. but on the other hand I think that it can break somehow the promise of the project. I mean that if someone deploys a Safe implementation of the memecoin using a know classhash of UnruggableMemecoin, he can still deploy with malicious values for the AMM and use a malicious AMM.
It will be hard to manage environments but I think we should consider hardcoding some stuff, eventually fill them at compile time with extensions, similar to what Dojo is doing.
We need to discuss about configuration / environments etc.